dask/array/core.py

```diff
@@ -2292,7 +2330,14 @@ def from_array(x, chunks='auto', name=None, lock=False, asarray=True, fancy=True
                     dtype=x.dtype)
         dsk[original_name] = x
 
-    return Array(dsk, name, chunks, dtype=x.dtype)
+    # Workaround for TileDB, its indexing is 1-based,
+    # and doesn't seems to support 0-length slicing
+    if hasattr(x, '_ctx_'):
+        return Array(dsk, name, chunks, dtype=x.dtype)
```

Comment:
This seems brittle.  Are there better checks that we can perform here to pin this down to TileDB more specifically?  Perhaps something like `'tiledb' in str(type(x).__name__) and hasattr(...)`

Reply:
I'll try that, I couldn't think of anything dead certain. Thanks for the suggestion.